extensions-jvm/asn1/deployment/src/main/java/org/apache/camel/quarkus/component/asn1/deployment/Asn1Processor.java

```diff
@@ -37,7 +37,7 @@ FeatureBuildItem feature() {
     /**
      * Remove this once this extension starts supporting the native mode.
      */
-    @BuildStep(onlyIf = NativeBuild.class)
+    @BuildStep(onlyIf = NativeOrNativeSourcesBuild.class)
     @Record(value = ExecutionTime.RUNTIME_INIT)
     void warnJvmInNative(JvmOnlyRecorder recorder) {
```

Comment:
In case of warnings like here, it could remain `NativeBuild` but for the sake of uniformity, I understand that it is changed for `NativeOrNativeSourcesBuild`

Reply:
IMO we should stick to it in order to warn also when native-sources is used.

Reply:
@essobedo Both side, we might be good to merge. Worst case, we could experiment and fix on main.